model/task/task.go

```diff
@@ -1220,27 +1229,59 @@ func GenerateNotRun() ([]Task, error) {
 	}))
 }
 
-// SetGeneratedJSON sets JSON data to generate tasks from.
-func (t *Task) SetGeneratedJSON(json []json.RawMessage) error {
-	if len(t.GeneratedJSONAsString) > 0 {
+// SetGeneratedJSON sets JSON data to generate tasks from. If the generated JSON
+// files have already been stored, this is a no-op.
+func (t *Task) SetGeneratedJSON(files GeneratedJSONFiles) error {
+	if len(t.GeneratedJSONAsString) > 0 || t.GeneratedJSONStorageMethod != "" {
```

Comment:
Yes, the expected behavior when populating the generated JSON (which the pre-change `len(GeneratedJSONAsString) > 0` check previously did) is that it inserts it (into the DB or S3) and then sets the storage method, and after it's set, generate.tasks is not allowed to change the generated JSON (because the task generate.tasks input is not allowed to change). If it's stored in the DB, checking that either `GeneratedJSONAsString` or `GeneratedJSONStorageMethod` is set prevents changing it. If it's stored in S3, then `GeneratedJSONStorageMethod` is the only flag we have to know that it's already been stored elsewhere.